store/mongo/datastore_mongo_test.go

```diff
@@ -728,3 +728,41 @@ func TestMongoDeleteDeviceAuthsByDevice(t *testing.T) {
 	err = dbstore.DeleteDeviceAuthByDevice(ctx, "0004")
 	assert.EqualError(t, err, store.ErrNotFound.Error())
 }
+
+func TestMongoInsertAndGetDeviceAuthsByIdentityData(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping TestMongoGetDevices in short mode.")
+	}
+
+	db.Wipe()
+	session := db.Session()
+	defer session.Close()
+
+	ctx := context.Background()
+	tenCtx := identity.WithContext(ctx, &identity.Identity{
+		Subject: "foo",
+		Tenant:  "bar",
+	})
+
+	dbstore := NewDataStoreMongoWithSession(session)
+
+	authSets, err := dbstore.GetDeviceAuthsByIdentityData(tenCtx, "{'foo': 'bar'}")
+
+	assert.NoError(t, err)
+	assert.Equal(t, 0, len(authSets))
+
+	dev := model.DeviceAuth{DeviceIdentity: "{'foo': 'bar'}"}
+
+	dev.Status = model.DevStatusPreauthorized
+	dev.Key = "foo-key"
+	now := time.Now()
+	dev.RequestTime = &now
+
+	err = dbstore.InsertDeviceAuth(tenCtx, &dev)
+	assert.NoError(t, err)
+
+	authSets, err = dbstore.GetDeviceAuthsByIdentityData(tenCtx, "{'foo': 'bar'}")
```

Comment:
consider verifying with a bare-bones `mgo.Find`.
my take on it: test one thing at a time, so that errors in 2 of our methods don't cancel each other.
we rely on 3rd party code as 100% correct, so this is what we use to verify our code.

Reply:
After discussion keeping this version. Adding second device that should not be found by the test as suggested by kjaskiewicz.